src/list.js

```diff
@@ -130,7 +130,7 @@ class TypeInferedList extends BaseImmutableList {
   }
 
   get(index, notSetValue) {
-    return this[$store] ? this[$store].get(index, notSetValue) :
+    return this[$store] ? this[$store].get(parseInt(index), notSetValue) :
```

Comment:
I guess my feeling is, when in doubt, keep the behavior we had already. If this was a from-scratch thing, I would want it to be an error.